### ISSUE TITLE
Locating LAI Calculation

### ISSUE THREAD

**Issue by afsharmehdi** [APSIMInitiative/APSIMClassic#2200]
Dear APSIM Community,

I am reaching out to request assistance in locating the specific codes and steps used for the simulation of Leaf Area Index (LAI) in APSIM Canola.

I have been searching through the APSIM Classic source code on GitHub, but have been unable to locate the specific files and directories involved in the calculation of LAI for canola. I am trying to write these scripts in R environment. I have been done a lot of efforts to emulate APSIM in R, but was unable to do this for LAI. I followed the papers and documents for better understanding of this issue. But so far was unable to move forward.

If possible, could someone provide me with a specific path or directory within the APSIM Classic source code where the relevant code can be found? Alternatively, if there is a resource or documentation available that outlines the steps involved in the LAI calculation, I would greatly appreciate any guidance in that regard.

**Comment by peter-devoil** [comment]
Yes, it's hard to find the model structure from source code - even harder to replicate it in another framework. You'll find Bangyou undertook a [partial re-implementation](https://apsiminitiative.github.io/APSIMWheatPhenology/) for his phenology exercise which might demontrate the scope of the problem.

The starting point in all apsim modules is its [xml formatted .ini file](https://github.com/APSIMInitiative/APSIMClassic/blob/master/Model/Canola.xml#L72) which says the leaf organ is a generic class (there are other leaf types), which exposes a state variable LAI available to the system [here](https://github.com/APSIMInitiative/APSIMClassic/blob/master/Model/Plant/Leaf/GenericLeaf.cpp#L120). The process to calculate LAI is to firstly calculate [a delta](https://github.com/APSIMInitiative/APSIMClassic/blob/master/Model/Plant/Leaf/GenericLeaf.cpp#L414) which is accumulated at the end of the day.
